qos/evm/endpoint.go

```diff
@@ -94,29 +97,44 @@ func (e endpoint) getBlockNumber() (uint64, error) {
 	return *e.parsedBlockNumberResponse, nil
 }
 
-func (e *endpoint) validateArchivalCheck(archivalBalance string, endpointAddr protocol.EndpointAddr) error {
-	if e.archivalBalance == "" {
+// validateArchivalCheck validates that the endpoint has returned an archival balance for the perceived block number.
+
+// If the archival check is not enabled for the service, this will always return a nil error.
+func (e *endpoint) validateArchivalCheck(archivalState archivalState) error {
```

Comment:
Not a blocker: a TODO that we will need to move these checks to the ServiceState struct rather than passing the state to each endpoint.

Reply:
I've added a todo in `state.go`'s `ValidateEndpoint` method:
```go
	// TODO_TECHDEBT(@commoddity): move the endpoint validation methods to the service state
	// and pass them the endpoint rather than passing the service state to each endpoint method.
```
Does this capture your intention?